src/coreclr/vm/dllimport.cpp

```diff
@@ -3167,6 +3167,40 @@ HRESULT NDirect::HasNAT_LAttribute(IMDInternalImport *pInternalImport, mdToken t
     return S_FALSE;
 }
 
+namespace
+{
+    bool IsSwiftIntrinsicValueType(TypeHandle hnd)
+    {
+        STANDARD_VM_CONTRACT;
+
+        _ASSERTE(hnd.IsValueType());
+        MethodTable* pMT = hnd.GetMethodTable();
+        if (!pMT->IsIntrinsicType())
+            return false;
+
+        LPCUTF8 namespaceName;
+        LPCUTF8 className = pMT->GetFullyQualifiedNameInfo(&namespaceName);
+
+        // We treat the .NET vector types as projections of the Swift SIMD intrinsic types.
+        if ((strcmp(className, "Vector512`1") == 0) || (strcmp(className, "Vector256`1") == 0) ||
+            (strcmp(className, "Vector128`1") == 0) || (strcmp(className, "Vector64`1") == 0))
+        {
+            _ASSERTE(strcmp(namespaceName, "System.Runtime.Intrinsics") == 0);
+            return true;
+        }
+
+        // The SwiftSelf type is an intrinsic type that represents the Swift self register.
+        if (strcmp(className, "SwiftSelf") == 0)
+        {
+            _ASSERTE(strcmp(namespaceName, "System.Runtime.InteropServices.Swift") == 0);
+            return true;
+        }
+
+        // We don't need to treat the SwiftError type as an intrinsic here as it should always be passed
+        // as a SwiftError* parameter.
```

Comment:
This is an important consideration and something we are going to need to have an intelligent analyzer for. Please add this comment in the managed type system.

Reply:
With our changed approach for handling structs (the JIT/VM will handle more of the support, exact shape TBD, instead of blocking it), this PR will have much of this checking removed.